### ISSUE TITLE
Support tree service

### ISSUE THREAD

**Issue by KirillovDenis** [nspcc-dev/restic#7]
It seems we should consider supporting tree service to get rid of the `SEARСH` operations. 

/cc @alexvanin @realloc

**Comment by alexvanin** [comment]
Should we care about compatibility with S3 Gateway tree service structure? I think we **can**, but it is **not necessary**, in general.

I would go with PoC branch, but it won't be merged until NeoFS supports tree service in master branch and in the public network.

**Comment by KirillovDenis** [comment]
Should we wait until the tree service is supported in the SDK?

**Comment by alexvanin** [comment]
Yeah, it is quite ugly to fetch tree service client from the source code of neofs-node. We don't want to import the storage node code here.
